### ISSUE TITLE
Issue: -P seems to be ignored 

### ISSUE THREAD

**Issue by gdanov** [nektos/act#796]
how to reproduce:
- on osx `brew install act`
- run with -j <some job> that has `runs-on: ubuntu-20.04`
- on first run act asks about the image. I choose "large"
- act ignores my choice and seems to use the defaults it set in ~/.actrc (ubuntu-18.04=nektos/act-environments-ubuntu:18.04 )
- I add my ubuntu-20 choice to ~/.actrc so that it looks like this `-P ubuntu-18.04=nektos/act-environments-ubuntu:18.04 ubuntu-20.04=ghcr.io/catthehacker/ubuntu:full-20.04`
- re-run act but same result
- I remove the ubuntu-18 target so that ~/.actrc looks like `-P ubuntu-20.04=ghcr.io/catthehacker/ubuntu:full-20.04`
- act downloads the image I want this time

**Comment by catthehacker** [comment]
correct syntax is
```
-P ubuntu-18.04=nektos/act-environments-ubuntu:18.04 -P ubuntu-20.04=ghcr.io/catthehacker/ubuntu:full-20.04
```

**Comment by gdanov** [comment]
it was completely ignored the fact that at start-up my entry is ignored and wrong environment is used.

**Comment by catthehacker** [comment]
Nothing was ignored, latest `act` release doesn't include my images (https://github.com/nektos/act/pull/766) and only `ubuntu-18.04` was available before that.

**Comment by gdanov** [comment]
let me try again: 

when I provide `-P ubuntu-18.04=nektos/act-environments-ubuntu:18.04 -P ubuntu-20.04=ghcr.io/catthehacker/ubuntu:act-20.04` in the ~/.actrc file

and run a job that requires ubuntu-20 (regular is ok)

I see `ImageOS:ubuntu20` in the logs

BUT the build fails with 
`::error::Unable to locate executable file: docker. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.`

If I have only the ubuntu-20 -P setting then the job works.

**Comment by catthehacker** [comment]
No idea, but you could follow the issue template and then I might be able to find out.

**Comment by catthehacker** [comment]
`.actrc` is newline delimited so you need to have
```
-P ubuntu-18.04=nektos/act-environments-ubuntu:18.04
-P ubuntu-20.04=ghcr.io/catthehacker/ubuntu:act-20.04
```

**Comment by gdanov** [comment]
Thanks. That works. So, basically it seems that 

when **no** -P is provided for non-default platform required in action's `run-on:`

then act silently uses the default one?

**Comment by catthehacker** [comment]
Yes, that's correct
